MoreLinq/FillForward.cs

```diff
@@ -112,10 +112,10 @@ static IEnumerable<T> FillForwardImpl<T>(IEnumerable<T> source, Func<T, bool> pr
             {
                 if (predicate(item))
                 {
-                    yield return seed is (true, { } someSeed)
+                    yield return seed is (true, var theSeed)
```

Comment:
```suggestion
                    yield return (seed, fillSelector) switch
                    {
                      ((true, var theSeed), { } f) => f(theSeed),
                      ((true, var theSeed), _) => theSeed,
                      _ => item,
                    };
```

Reply:
I'm going to skip this suggestion since I want the changes for this bug fix to be as minimal and obvious as possible. Changing to a `switch` expression with pattern-matching is a separate issue of whether it reads better than nested conditionals. It belongs in a separate issue/PR.